src/pages/workspaces/workspace/Data.js

```diff
@@ -544,9 +550,28 @@ const DataTableActions = ({ workspace, tableName, rowCount, onUpdateSuccess }) =
     }),
     renaming && h(RenameTableModal, {
       onDismiss: () => setRenaming(false),
-      onUpdateSuccess,
+      onUpdateSuccess: onRenameTable,
       namespace, name,
       selectedDataType: tableName
+    }),
+    deleting && h(DeleteConfirmationModal, {
+      objectType: 'table',
+      objectName: tableName,
+      onDismiss: () => setDeleting(false),
+      onConfirm: async () => {
+        setDeleting(false)
```

Comment:
I was thinking that the `setBusyState(false)` after successfully deleting a table would cause a state update on an unmounted component because deleting the table removes the table and this menu from the sidebar. But I tried it with `withErrorReporting` and `withBusyState` and there was no warning.